GoogleCloudExtension/GoogleCloudExtension/Accounts/WindowsCredentialsStore.cs

```
@@ -125,13 +127,43 @@ public string GetStoragePathForInstance(Instance instance)
             return Path.Combine(s_credentialsStoreRoot, instancePath);
         }
 
+        /// <summary>
+        /// Attempts to load and decrypt the credentials stored in <paramref name="path"/> and returns an
+        /// <seealso cref="WindowsInstanceCredentials"/> instance with the information stored in the file. If
+        /// the file cannot be loaded or decrypted it will return null.
+        /// 
+        /// Note: The function will attempt to delete the file if it cannot be decrypted, this typically means that
+        /// the user's key is no longer valid. The function does not attempt to delete the file in case of a <see cref="IOException"/>
+        /// since that will probably also throw again.
+        /// </summary>
         private WindowsInstanceCredentials LoadEncryptedCredentials(string path)
         {
-            var userName = GetUserName(path);
-            var encryptedPassword = File.ReadAllBytes(path);
-            var passwordBytes = ProtectedData.Unprotect(encryptedPassword, null, DataProtectionScope.CurrentUser);
+            try
+            {
+                var userName = GetUserName(path);
+                var encryptedPassword = File.ReadAllBytes(path);
+                var passwordBytes = ProtectedData.Unprotect(encryptedPassword, null, DataProtectionScope.CurrentUser);
 
-            return new WindowsInstanceCredentials { User = userName, Password = Encoding.UTF8.GetString(passwordBytes) };
+                return new WindowsInstanceCredentials { User = userName, Password = Encoding.UTF8.GetString(passwordBytes) };
+            }
+            catch (CryptographicException)
+            {
+                Debug.WriteLine($"Failed to decrypt credentials from: {path}");
+                try
+                {
+                    File.Delete(path);
+                }
+                catch (IOException)
```

Do we want to fail silently on `IOException`? That feels like the type of thing a user can fix, and therefore should be informed of.